webpack/docs.config.js

```diff
@@ -1,18 +1,47 @@
-import config from './webpack.config';
-import yargs from 'yargs';
+import _ from 'lodash';
+import webpack from 'webpack';
+import ExtractTextPlugin from 'extract-text-webpack-plugin';
+import baseConfig, { argv } from './base.config';
 
-const argv = yargs
-  .alias('p', 'optimize-minimize')
-  .alias('d', 'debug')
-  .option('port', {
-    default: '8080',
-    type: 'string'
-  })
-  .argv;
+const webpackDevServerAddress = `http://localhost:${argv.port}`;
+const cssSourceMap = argv.debug ? '?sourceMap' : '';
+const reactHot = argv.debug ? 'react-hot!' : '';
 
-export default config({
-  docs: true,
-  development: argv.debug,
-  optimize: argv.optimizeMinimize,
-  port: parseInt(argv.port)
+const entryFile = './docs/client.js';
+const devEntryBundle = [
+  'webpack/hot/dev-server',
+  `webpack-dev-server/client?${webpackDevServerAddress}`,
+  entryFile
+];
+
+baseConfig.plugins.push(new ExtractTextPlugin('[name].css'));
+if (argv.debug) {
+  baseConfig.plugins.push(new webpack.NoErrorsPlugin());
+}
+
+export default _.extend({}, baseConfig, {
+  entry: {
+    bundle: argv.debug ? devEntryBundle : entryFile
+  },
+
+  output: {
+    filename: '[name].js',
+    path: './docs-built/assets',
+    publicPath: argv.debug ? `${webpackDevServerAddress}/assets/` : '/assets/'
+  },
+
+  module: {
+    noParse: /babel-core\/browser/,
+    loaders: [
+      { test: /\.js/, loader: `${reactHot}babel`, exclude: /node_modules|Samples\.js/ },
+      { test: /Samples.js/, loader: `${reactHot}transform?brfs!babel` },
```

Comment:
One of the reasons I was doing a search for the loader I wanted to change was to allow changes in the base loader to cascade down. Since the base loader only has the one loader, how would you feel about including that as an export of the base and build these two on top of that one?